src/Components/Main.tsx

```diff
@@ -0,0 +1,199 @@
+  onPressClose: (value: boolean) => void;
+  onPressDetail: (value: boolean) => void;
+  visible?: boolean;
+  reduxActions: ReduxAction[];
+  rnRequests: RNRequest[];
+  clearAll: Function;
+}
+
+export const Main = (props: IProps) => {
+  const [requests, setRequests] = useState<Array<ILog>>([]);
+  const [searchQuery, setSearchQuery] = useState('');
+  const [source, setSource] = useState<SourceType>('ALL');
+  const [filter, setFilter] = useState<RequestMethod | 'ALL'>('ALL');
+  const [settingsVisible, setSettingsVisible] = useState(false);
+  const [deleteVisible, setDeleteVisible] = useState(false);
+  const onChangeSearch = (query: string) => setSearchQuery(query);
+  const showDialog = () => setSettingsVisible(true);
+  const hideSettingsDialog = () => setSettingsVisible(false);
+  const showDialogDelete = () => setDeleteVisible(true);
+  const hideDialogDelete = () => setDeleteVisible(false);
+  const hideDialogAndDelete = () => {
+    clearList();
+    hideDialogDelete();
+  };
+
+  useEffect(() => {
+    if (source === 'REDUX') {
+      setRequests(props.reduxActions.slice(0, maxRequests));
+    } else if (source === 'RNR') {
+      setRequests(props.rnRequests.slice(0, maxRequests));
+    } else if (source === 'NR') {
+      // setRequests(nRequests.reverse().slice(0, maxRequests));
+    } else {
+      setRequests([...props.reduxActions, ...props.rnRequests].slice(0, maxRequests));
+      // setRequests([...props.rnRequests, ...props.reduxActions, ...props.nRequests].reverse().slice(0, maxRequests));
+    }
+    return () => {};
+  }, [props.rnRequests, props.reduxActions]);
+
+  const filteredRequests = requests
+    .filter((element) => source === 'ALL' || element.type === source)
+    .filter(
+      (request) =>
+        request.type === 'REDUX' ||
+        (request.type !== 'REDUX' &&
+          (request.url?.toLowerCase().includes(searchQuery.toLowerCase()) ||
+            request.status === parseInt(searchQuery) ||
+            request.method === searchQuery) &&
+          (filter === 'ALL' || request.method.toLowerCase() === filter))
+    )
+    .reverse();
+
+  const clearList = () => {
+    props.clearAll();
+    setRequests([]);
+  };
+
+  return (
+    <>
+      <Appbar.Header style={styles.header}>
+        <IconButton color="white" icon="close" onPress={() => props.onPressClose(false)} />
+        <Appbar.Content title="Netwatch" />
+      </Appbar.Header>
+      <Surface style={styles.surface}>
+        <Searchbar
+          placeholderTextColor="white"
+          iconColor="white"
+          inputStyle={styles.searchBarInputStyle}
+          style={styles.searchBar}
+          placeholder={'Search'}
+          onChangeText={onChangeSearch}
+          value={searchQuery}
+        />
+        <IconButton
+          // style={{ marginRight: -8 }}
+          icon={() => <Icon color="white" size={24} name="delete"></Icon>}
+          size={24}
+          onPress={showDialogDelete}
+        ></IconButton>
```

Comment:
cannot use `/>` instead of `></IconButton>`?

Reply:
Done